### ISSUE TITLE
`<vector>`, etc.: avoid `_RERAISE` in favor of usual scope guards

### ISSUE THREAD

**Issue by AlexGuteniev** [microsoft/STL#2307]
`_RERAISE;` expands to `throw;`, that is it rethrowing an exception.

This is used for cleanups in exception case.

It is inconvenient for debugging, when the same exception shows up twice.
Change this construct to usual scope guards.

**Comment by StephanTLavavej** [comment]
Agreed; the inconsistency exists because `_RERAISE` is the original pattern and scope guards are the modern pattern, but we hadn't overhauled all occurrences.

This can be done incrementally, e.g. file-by-file.

**Comment by frederick-vs-ja** [comment]
- 2 in `<chrono>`
  - I guess scope guard is unsuitable since sometimes we need to change the exception.
- 10 in `<vector>`
  - There're nested try-catch-rethrow in the current implementation (now in `_Insert_counted_range`) which can't be directly converted to scope guards. We may need to restructure `_Insert_counted_range`.
    https://github.com/microsoft/STL/blob/0a2d59ed7bfe099199237090998b758c9b0b7403/stl/inc/vector#L1173-L1192
    https://github.com/microsoft/STL/blob/0a2d59ed7bfe099199237090998b758c9b0b7403/stl/inc/vector#L1198-L1216
- 1 in `<xiosbase>`
  - I don't think this is doable because iostream classes are designed to be _conditionally_ rethrowing.
